main/src/ca/uwaterloo/flix/language/phase/typer/RestrictableChooseConstraintGen.scala

```diff
@@ -75,99 +76,114 @@ object RestrictableChooseConstraintGen {
       case KindedAst.Expr.RestrictableChoose(false, exp0, rules0, tpe0, loc) =>
 
         // Get the enum symbols for the matched type
-        val enumSym = rules0.headOption.getOrElse(throw InternalCompilerException("unexpected empty choose", loc)).pat match {
-          case KindedAst.RestrictableChoosePattern.Tag(sym, _, _, _) => sym.sym.enumSym
+        rules0.headOption match {
+          case None => throw InternalCompilerException("unexpected empty choose", loc)
+          case Some(rule) => rule.pat match {
+            case RestrictableChoosePattern.Error(_, loc) => // If the pattern is an error we just return fresh error variables
+              (Type.freshError(Kind.Error, loc), Type.freshError(Kind.Error, loc))
```

Comment:
Do we use `Kind.Error` elsewhere like this? I thought `Kind.Error` is only when we don't know the kind(?)

Reply:
I think I just assumed that this was the right thing to do. I can turn it into a `Star` instead :)